app/Http/Controllers/TalksController.php

```diff
@@ -163,4 +146,52 @@ public function destroy($id)
 
         return Redirect::to('talks');
     }
+
+    public function archiveIndex()
+    {
+        $talks = $this->sortTalks(
+            Auth::user()->talks()->archived()->get()
+        );
+
+        return View::make('talks.archive')
+          ->with('talks', $talks)
+          ->with('sorted_by', $this->sorted_by);
+    }
+
+    public function archive($id)
+    {
+        Auth::user()->talks()->findOrFail($id)->archive();
+
+        Session::flash('message', 'Successfully archived talk.');
+
+        return Redirect::to('talks');
+    }
+
+    public function restore($id)
+    {
+        Auth::user()->talks()->findOrFail($id)->restore();
+
+        Session::flash('message', 'Successfully restored talk.');
+
+        return Redirect::to('archive');
+    }
+
+    private function sortTalks($talks)
+    {
+        switch (Input::get('sort')) {
+            case 'date':
+                $this->sorted_by = 'date';
+                $talks = $talks->sortByDesc('created_at');
```

Comment:
Can we just `return $talks->sortByDesc` and same below, instead of setting a temp var and then returning?